packages/backend-core/src/db/pouch.js

```diff
@@ -1,35 +1,36 @@
 const PouchDB = require("pouchdb")
 const env = require("../environment")
 
-exports.getCouchUrl = () => {
-  if (!env.COUCH_DB_URL) return
-
-  // username and password already exist in URL
-  if (env.COUCH_DB_URL.includes("@")) {
-    return env.COUCH_DB_URL
-  }
-
-  const [protocol, ...rest] = env.COUCH_DB_URL.split("://")
-
-  if (!env.COUCH_DB_USERNAME || !env.COUCH_DB_PASSWORD) {
-    throw new Error(
-      "CouchDB configuration invalid. You must provide a fully qualified CouchDB url, or the COUCH_DB_USER and COUCH_DB_PASSWORD environment variables."
-    )
+exports.getCouchInfo = () => {
```

Comment:
Great that everything now uses a consistent connection config 👍 

We could probably make this function more readable, I found the current flow (construct auth aware url -> destruct auth aware url -> fallback to env vars) a bit hard to understand, but I get why it's done like that so the lower half of the function operates on the same url structure. 

Rather than constructing / destructing the url, we could prefer the auth vars first and fallback to the url auth if provided. I found this to be more readable and could look like:

```js
/**
 * Return the couch db connection config. 
 * Optionally include the username and password if present.
 */
const getConectionConfig = () => {
  let url = env.COUCH_DB_URL
  let username
  let password
  let host
  const [ protocol, ...rest] = url.split("://")

  if (url.includes("@")) {
    const hostParts = rest.split("@")
    host = hostParts[1]
    const authParts = hostParts[0].split(":")
    username = authParts[0]
    password = authParts[1]
  } else {
    host = rest
  }

  return {
    url: `${protocol}://${host}`,
    auth: {
      username,
      password
    }
  }
}

/**
 * Construct the couch db connection configuration. 
 * Always prefer the authentication provided from environment variables, 
 * fallback to authentication provided by the url if not present.
 */
exports.getCouchInfo = () => {
  const url = getConectionConfig()

  let username
  let password

  if (env.COUCH_DB_USERNAME) {
    // set from env 
    username = env.COUCH_DB_USERNAME
  } else if (url.auth.username) {
    // set from url
    username = url.auth.username
  } else {
    throw new Error("CouchDB username not set")
  }

  if (env.COUCH_DB_PASSWORD) {
    // set from env
    password = env.COUCH_DB_PASSWORD
  } else if (url.auth.password) {
    // set from url
    password = url.auth.password
  } else {
    throw new Error("CouchDB password not set")
  }

  const authCookie = Buffer.from(`${username}:${password}`).toString("base64")

  return {
    url: url.url,
    auth: {
      username: username,
      password: password,
    },
    cookie: `Basic ${authCookie}`,
  }
}
```

Could also move the default of the localhost url out to the env like 
```js
COUCH_DB_URL: process.env.COUCH_DB_URL | "http://localhost:4005"
```